decision/value_functions/abstract.py

```diff
@@ -128,88 +135,76 @@ def get_number_of_location_indicators_and_state_features(
-            + scooter_inventory_indication
-            + battery_inventory_indication
-            + small_depot_degree_of_filling
-        )
-
-    def update_shifts_trained(self, shifts_trained: int):
-        self.shifts_trained = shifts_trained
-
-    @Decorators.check_setup
-    def convert_next_state_features(self, state, vehicle, action, time, cache=None):
+        # Create a flag for if it is a get next state features call
+        is_next_action = action is not None
         # Change location by swapping location indicator
         location_indicators = self.get_location_indicator(
-            action.next_location, len(state.locations)
+            action.next_location if is_next_action else vehicle.current_location.id,
+            len(state.locations),
         )
         # Fetch all normalized scooter state representations
         (
             normalized_deviation_ideal_state_positive,
             normalized_deviation_ideal_state_negative,
             normalized_deficient_battery,
         ) = ValueFunction.get_normalized_lists(
-            state, cache, current_location=vehicle.current_location.id, action=action,
+            state,
+            cache,
+            current_location=vehicle.current_location.id if is_next_action else None,
+            action=action,
         )
         # Inventory indicators adjusting for action effects
         scooter_inventory_indication = self.get_inventory_indicator(
             helpers.zero_divide(
                 len(vehicle.scooter_inventory)
-                + len(action.pick_ups)
-                - len(action.delivery_scooters),
+                + (
+                    len(action.pick_ups) - len(action.delivery_scooters)
+                    if is_next_action
+                    else 0
+                ),
                 vehicle.scooter_inventory_capacity,
             )
         )
 
         battery_inventory_indication = self.get_inventory_indicator(
             helpers.zero_divide(
                 vehicle.battery_inventory
-                - len(action.battery_swaps)
-                - len(action.pick_ups),
+                - (
+                    len(action.battery_swaps) + len(action.pick_ups)
+                    if is_next_action
+                    else 0
+                ),
                 vehicle.battery_inventory_capacity,
             )
         )
         # Depot state
         small_depot_degree_of_filling = ValueFunction.get_small_depot_degree_of_filling(
             time, state
         )
+        # Add one hot encoding for the location of the other vehicles
+        other_vehicles_locations = [
+            indicator
+            for other_vehicle in state.vehicles
+            if other_vehicle.id != vehicle.id
+            for indicator in self.get_location_indicator(
+                other_vehicle.current_location.id,
+                len(state.locations),
+                location_repetition=1,
+            )
+        ]
```

Comment:
Thought we could just have a list of length ```len(state.clusters)``` and one-hot that list with the other vehicles locations, Occam's razor. With this implementation, we will get a lot of zeros and a lot of unnecessary weights. Bad explanation, but I think we should discuss tomorrow the implementation I suggested.

Reply:
One hot encoding creates a lot of zeros, [wiki page](https://en.wikipedia.org/wiki/One-hot). This is not a good reference to Occam's razor, but simply a misunderstanding of one-hot encoding. Both the explanation and the comment is bad. Please consider reading up on the theory before leaving these kind of nonsense. If you have another way of creating the one-hot encoding, or a new way to encode the other vehicles location please provide it here. I am quite tired when writing this comment, so I could be wrong. Sorry about that.